internal/app/transport/watchv2.go

```diff
@@ -43,10 +49,22 @@ func (w *watchV2) Send(s Response) error {
 	if w.closed {
 		return nil
 	}
+
+	// Drop the older response currently in the transport queue, so that we
```

Comment:
I wanted to brainstorm an alternative approach based on the version of gcp we use https://github.com/envoyproxy/go-control-plane/blob/53deebd8777dd83921cb83a1975e7ec13fb8bb04/pkg/server/sotw/v3/server.go#L200

If instead of deque/enqueue operation, if we closed the channel , then the stream will be terminated and the envoy sidecar will create a fresh stream. I think this will help keep the implementation simple. wdyt?

Reply:
Another aspect to think about is : The channel drop happens when the transport is blocked, which can only happen when there is stress in the system(cpu/network). One way to apply backpressure in such situation would be to close the stream so that the sidecar will recreate the stream(which can load balance and hit another pod).

Reply:
The PR approach is fine too and we can go ahead with it, just wanted to make sure we add extra code with the documented reasons.

Reply:
I think it depends if we want to cause thrashing at the transport layer (clients -> control plane) or have the control plane alleviate that burden. It depends on the setup of the clients IMO. Imagine a scenario if the clients all reside in a single environment (ex: a Kubernetes cluster) with xds-relay in a different environment. If the client environment is degraded and we started closing watches, we'll see a churn of new connections being established to the control plane. With the current approach, we essentially hold the connection until the clients are ready to receive updates again.

Reply:
'we'll see a churn of new connections being established to the control plane` there will be stream churn and not connection. The current approach is fine..i was trying to see if we can achieve same effect with less code